### ISSUE TITLE
sab - 1.2.x -- article cache

### ISSUE THREAD

**Issue by thezoggy** [sabnzbd/sabnzbd#685]
looking through my logs I dont really see the article cached used.. and when i view it via the glitter popup.. (not a great place to find or ask someone to check/recheck).. it looks like its rarely used... no matter the size I give it. 

thus, I suggest that we add the peak article cache used to the report at the end so we can list.. thus people could actually see if its being used and how much.. thus can better tune that setting. I see some people that dont set it.. some that set it to 5G..  I've tried doing 256M and 512M but see no difference.. also does it require rebooting sab inbetween for it to take effect?

not sure if we track it.. if we do, we could list total / peak / avg values ? if we dont, just track highest during the d/l? (which is skewed if you have multiple d/ls going?)

**Comment by thezoggy** [comment]
reason i bring this up is that i had a debate with someone recently about the size of articles posted and appropriate values in sab to use..  and how article cache works..  

some people think if you put 5G it reserves 5G of memory like a ram disk and puts articles there when needed.. some people view it as a virtual swap that grows  upto that limit.. etc .. 

https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/articlecache.py#L190

also somewhat hard to test when the LOG_ALL option is only via startup argument `--log-all`.. might be nice to just add a special switch for that as well? also prob should move some of those downloader decode stuff to LOG_ALL as well to reduce disk io for the masses?

**Comment by Safihre** [comment]
You'll see difference when you download jobs where the individual RAR's are +100MB each, since every RAR will be kept in memory until it's all done and ready to be written. But with TV shows the sizes are usually 15-30MB, so that's all the memory usage you'll see for decoded articles.
On slower PC's its actually the decoder queue, with un-decoded items, that takes over memory usage, but even this should stay limited.
When the limit is reached, you'll see in the log ```Delaying```, where the decoder pauses the downloader in order to lower the amount of articles queued for the decoder.

So basically any cache will help, but you definitely don't need more than 1G. Probably more than 500M is already excessive. 
While I understand the informative value of noting max/avg, I'm not completely sure if it's worth it?

**Comment by thezoggy** [comment]
personally id love to see this graph like bandwidth to see whats going on..  to show people if they are being disk io limited or what..     but this is mainly for the dev in me not for user experience

**Comment by Safihre** [comment]
How can we make it a bit more clear for the user maybe?

**Comment by thezoggy** [comment]
noticed that the article cache is shown on the config base page.. which is much nicer to refresh than the glitter modal.. i can just use this while something is d/ling to get a sense of how much is being used

**Comment by Safihre** [comment]
Now the article cache in the Status Window gets update with every Queue refresh to get some more info.

Think that should give enough info :)
